operator/internal/manifests/internal/config/options.go

```diff
@@ -39,6 +40,100 @@ type Address struct {
 type Ruler struct {
 	Enabled               bool
 	RulesStorageDirectory string
+	EvaluationInterval    string
+	PollInterval          string
+	AlertManager          *AlertManagerConfig
+	RemoteWrite           *RemoteWriteConfig
+}
+
+// AlertManagerConfig for ruler alertmanager config
+type AlertManagerConfig struct {
+	Hosts          string
+	ExternalURL    string
+	ExternalLabels map[string]string
+	EnableV2       bool
+
+	// DNS Discovery
+	EnableDiscovery bool
+	RefreshInterval string
+
+	// Notification config
+	QueueCapacity      int32
+	Timeout            string
+	ForOutageTolerance string
+	ForGracePeriod     string
+	ResendDelay        string
+}
+
+// RemoteWriteConfig for ruler remote write config
+type RemoteWriteConfig struct {
+	Enabled        bool
+	RefreshPeriod  string
+	Client         *RemoteWriteClientConfig
+	Queue          *RemoteWriteQueueConfig
+	RelabelConfigs []RemoteWriteRelabelConfig
+}
+
+// RemoteWriteClientConfig for ruler remote write client config
+type RemoteWriteClientConfig struct {
+	Name            string
+	URL             string
+	RemoteTimeout   string
+	Headers         map[string]string
+	ProxyURL        string
+	FollowRedirects bool
+
+	// Authentication
+	BasicAuthUsername string
+	BasicAuthPassword string
+	HeaderBearerToken string
+}
+
+// RemoteWriteQueueConfig for ruler remote write queue config
+type RemoteWriteQueueConfig struct {
+	Capacity          int32
+	MaxShards         int32
+	MinShards         int32
+	MaxSamplesPerSend int32
+	BatchSendDeadline string
+	MinBackOffPeriod  string
+	MaxBackOffPeriod  string
+}
+
+// RemoteWriteRelabelConfig for ruler remote write relabel configs.
+type RemoteWriteRelabelConfig struct {
+	SourceLabels []string
+	Separator    string
+	TargetLabel  string
+	Regex        string
+	Modulus      uint64
+	Replacement  string
+	Action       string
+}
+
+// SourceLabelsString returns a string array of source labels.
+func (r RemoteWriteRelabelConfig) SourceLabelsString() string {
+	var sb strings.Builder
+	sb.WriteString("[")
+	for i, labelname := range r.SourceLabels {
+		sb.WriteString(fmt.Sprintf(`"%s"`, labelname))
+
+		if i != len(r.SourceLabels)-1 {
+			sb.WriteString(",")
+		}
+	}
+	sb.WriteString("]")
+
+	return sb.String()
+}
+
+// SeparatorString returns the user-defined separator or per default semicolon.
+func (r RemoteWriteRelabelConfig) SeparatorString() string {
+	if r.Separator == "" {
+		return ";"
```

Comment:
This might be an edge-case, but I don't think an empty separator should be disallowed. Someone might just want to concatenate two different values to form a new one.